src/engine.ml

```diff
@@ -535,16 +567,18 @@ let unpad block_size cs =
   else
     Error (`Msg "bad padding")
 
-let data_out (ctx : keys) compress protocol rng key data =
-  (* output is: packed_id 0xfa data, then wrap openvpn partial header
+let out ?ts (ctx : keys) compress rng data =
+  (* output is: packed_id <ts_opt> 0xfa data, then wrap openvpn partial header
      ~~> well, actually take a random IV, pad and encrypt,
      ~~> prepend IV to encrrypted data
      --> hmac and prepend hash *)
-  let hdr = Cstruct.create (4 + if compress then 1 else 0) in
+  let hdr_len = 4 + (if compress then 1 else 0) + (match ts with None -> 0 | Some _ -> 4) in
```

Comment:
```suggestion
  let hdr_len = 4 + (match ts with None -> 0 | Some _ -> 4) + (if compress then 1 else 0) in
```
Since the timestamp comes before the compression tag, this ordering makes more sense to my brain :)